src/mount/conf.cc

```diff
@@ -66,26 +66,20 @@ extern "C" void mount_ceph_get_config_info(const char *config_file,
 	continue;
     }
 
-    std::string addr;
-    addr += eaddr.ip_only_to_str();
-    addr += ":";
-    addr += std::to_string(eaddr.get_port());
-    /* If this will overrun cci_mons, stop here */
-    if (monaddrs.length() + 1 + addr.length() + 1 > sizeof(cci->cci_mons))
-      break;
-
-    if (first)
-      first = false;
-    else
-      monaddrs += ",";
-
-    monaddrs += addr;
+    if (oss.tellp() > 0) {
+      oss << ",";
+    }
+    oss << eaddr.get_sockaddr();
   }
 
-  if (monaddrs.length())
+  monaddrs = oss.str();
+  if (monaddrs.length() + 1 > sizeof(cci->cci_mons)) {
+    mount_ceph_debug("Monitor addresses are too long (length parsed: %d, max length: %d)\n", monaddrs.length() + 1, sizeof(cci->cci_mons));
```

Comment:
Can we do something smarter here like trim the list of monitors down to maybe 2 or 3? Also I don't see any error thrown for this condition?

Reply:
This is what the original code did and I suggested getting rid of that.  The mount helper shouldn't mess with the monitor list IMO.

The error is thrown later:
```
	if (!cmi.cmi_mons) {
		fprintf(stderr, "unable to determine mon addresses\n");
		retval = EX_USAGE;
		goto out;
	}
```

Reply:
@batrick @idryomov I'm okay with either implementation. Will leave the philosophical debate to you 😄

Reply:
@batrick @idryomov just wanted to check on this! Do we want to add additional smartness or keep it as is?

Reply:
Up to @jtlayton and @batrick...

Reply:
> The mount helper shouldn't mess with the monitor list IMO.

Alright, agreed. @ben-z I still don't see the program exit if it encounters these fatal conditions though. Just debugging...

Reply:
Hrm, maybe that if condition above needs to check not just for NULL but also for emptiness?